tests/library_app/models.py

```diff
@@ -3,31 +3,59 @@
 from django.utils.translation import gettext_lazy as _
 
 from garnett import fields
-from garnett.utils import get_current_language
+from garnett.translatedstr import TranslatedStr
+from garnett.utils import get_languages, get_current_language
 
 
 def validate_length(value):
     if len(value) < 3:
         raise ValidationError(_("Title is too short"))
 
 
-def title_fallback(field, obj):
-    current_lang = get_current_language()
-    if obj.translations.title.items():
-        lang, value = list(obj.translations.title.items())[0]
-        return f"{value} (Book title unavailable in {current_lang}, falling back to {lang})"
-    else:
-        return "No translations available for this book"
+class TitleTranslatedStr(TranslatedStr):
+    def get_fallback_text(content):
+        if content.items():
+            for lang in get_languages():
+                if lang.language in content:
+                    value = content[lang.language]
+                    return (lang, f"{value}")
+        else:
+            return None, "No translations available for this book"
+
+    def __html__(self) -> str:
```

Comment:
This seems similar to the above code, is there duplication?

Reply:
Its a demo function to showcase stuff.